src/main/java/io/strimzi/kafka/metrics/KafkaPrometheusMetricsReporter.java

```diff
@@ -48,10 +52,28 @@ public KafkaPrometheusMetricsReporter() {
         this.registry = registry;
     }
 
+    /**
+     * Sets the prefix to be used for metric names. This is always called before addMetric/removeMetric
+     *
+     * @param prefix The prefix to set.
+     */
+    public void setPrefix(String prefix) {
+        this.prefix = PrometheusNaming.prometheusName(prefix);
+    }
+
+    /**
+     * This method is used to get the prefix that is used for metric names.
+     *
+     * @return The prefix used for metric names.
+     */
+    public String getPrefix() {
+        return prefix;
+    }
```

Comment:
We can remove these 2 methods and just set the `prefix` field in `contextChange()`.

Reply:
Removed